Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM ruby:2.3.1-alpine
+MAINTAINER Code Climate <hello@codeclimate.com>
+
+RUN apk --update add git build-base autoconf automake libtool ruby-dev ruby-bundler snappy && \
+  rm -fr /usr/share/ri
+
+ENV KAFKA_SRC=http://apache.cs.utah.edu/kafka/0.8.2.1/kafka_2.10-0.8.2.1.tgz
+ENV KAFKA_PATH=/usr/src/kafka
+
+RUN curl "$KAFKA_SRC" | tar xvzf - && mv kafka_* "$KAFKA_PATH"
+
+WORKDIR /usr/src/poseidon/
+
+COPY .git/ /usr/src/poseidon/
+COPY Gemfile* /usr/src/poseidon/
+COPY poseidon.gemspec /usr/src/poseidon/
+COPY lib/poseidon/version.rb /usr/src/poseidon/lib/poseidon/
+
+RUN bundle install -j 4
+
+COPY . /usr/src/poseidon/
```

Comment:
I don't see any strong advantages to this: the COPY step is very fast (milliseconds) since the repo's not too big. I think there are potentially confusing disadvantages as well: you could change dependency info & get newer files, but be running without updated dependencies. And it's an extra flag to remember every time you might want to run something against the image directly for debugging (maybe a console or shell or whatever), and would put it somewhat at odds where we usually do COPY. The existing `make test` task will work just as well locally as on CI, so I guess I just don't see this change as that useful.